Lib/test/test_compile.py

```diff
@@ -444,6 +444,10 @@ def f():
         self.assertIn("_A__mangled_mod", A.f.__code__.co_varnames)
         self.assertIn("__package__", A.f.__code__.co_varnames)
 
+    def test_condition_expression_with_dead_blocks_compiles(self):
+        # See gh-113054
+        compile('if 5 if 5 else T:y', '<eval>', 'exec')
```

Comment:
Nit: although it's slightly less minimal, I'd find it clearer to understand exactly what's being tested here if you added some parens

```suggestion
        compile('if (5 if 5 else T): 0', '<eval>', 'exec')
```